doc/source/whatsnew/v1.4.3.rst

```diff
@@ -24,6 +24,7 @@ Fixed regressions
 - Fixed regression in :func:`read_csv` with ``index_col=False`` identifying first row as index names when ``header=None`` (:issue:`46955`)
 - Fixed regression in :meth:`.DataFrameGroupBy.agg` when used with list-likes or dict-likes and ``axis=1`` that would give incorrect results; now raises ``NotImplementedError`` (:issue:`46995`)
 - Fixed regression in :meth:`DataFrame.resample` and :meth:`DataFrame.rolling` when used with list-likes or dict-likes and ``axis=1`` that would raise an unintuitive error message; now raises ``NotImplementedError`` (:issue:`46904`)
+- Fixed regression in :func:`assert_index_equal` when ``check_order=False`` and :class:`Index` has extension or object dtype (:issue:`47207`)
```

Comment:
@phofl FYI ``:func:`assert_index_equal` `` doesn't render. probably needs to be ``:func:`testing.assert_index_equal` ``.

no need to follow-up as can be done with other cleanups before release.

Reply:
Thanks, will try to do correctly in the future